homework/list.js

```diff
@@ -1,79 +1,85 @@
 'use strict';
 
-(function() {
+function List() {
+	this._listNoteContent = [];
+	this._wrapper = document.querySelector('#note-list-wrapper');
+}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+List.prototype = {
+	start : function () {
+		this.fetchList().then(function (data){
+			this.updateList(data);
+		}.bind(this)).then(
+			this.drawList.bind(this)
+		).then(
+			this.preloadFirstNote.bind(this)
+		);
+		window.addEventListener('click', (function (event) {
+			this.onNoteOpen(event);
+		}).bind(this));
+	},
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+	onNoteOpen : function (event) {
```

Comment:
Well it's a convention to prefix with `on` to handle the corresponding event, rather than to fire it. So, you should rename this function as `onClick` and to dispatch the `note-open`.